### PR TITLE
Fast Query System

### PR DESCRIPTION
**Pull request by drg101**
Changed a lot of things. Tracts are very, very fast now. Counties need to be simplified on the server-side, but when they are, we will have crazy fast queries.

switched geometryLoader to be a worker that queries based on geohashes. 

@pierce-smith1 I had to change some queryWorker code to get things working

### CODE REVIEW

File: src/iframe/js/library/GeometryLoaderJob.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
```
Comment (drg101) [inline src/iframe/js/library/GeometryLoaderJob.js:1]: btw, this file isn't used, but I will use it in a change very soon.

File: src/iframe/map2.html [hunk 2]
```diff
@@ -118,6 +117,7 @@
     <script src="js/ui/modeling/components/ModelResolution.js"></script>
 
     <!-- Frame JS, Load Last! -->
+    <script src="js/frame-specific/cool-guy.js"></script>
```
Comment (Marc-9) [inline src/iframe/map2.html]: this file does not exist

File: src/iframe/js/library/smartQuerier.js [hunk 3]
```diff
@@ -212,6 +212,7 @@ class SmartQuerier {
       * @method addToCache
       */
     addToCache(collection, data) {
+        return; //disabling this for until we can standardize things between query libs
```
Comment (Marc-9) [inline src/iframe/js/library/smartQuerier.js:215]: just comment out the code so we don't have errors in console

File: src/iframe/js/library/smartQuerier.js [hunk 4]
```diff
@@ -133,6 +132,7 @@ class SmartQuerier {
       */
     wrapResponseCallback(collection, callback) {
         let onResponseCallback = (response) => {
+            console.log("hereee")
```
Comment (Marc-9) [inline src/iframe/js/library/smartQuerier.js:135]: remove

File: src/iframe/js/library/boundsToGISJOIN.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    buckets: null, //loaded at config
+    geohashResolution: 3,
+
+    config: function (collection) {
+        const jURL = collection === "tract_geo_140mb" ? pathsToBuckets.tracts : pathsToBuckets.counties;
+        fetch(jURL)
+            .then(response => response.json())
+            .then(data => {
+                BoundsToGISJOIN.buckets = data
+                console.log(`Loaded ${collection} buckets`)
```
Comment (Marc-9) [inline src/iframe/js/library/boundsToGISJOIN.js:16]: Remove

**Review by Marc-9 — CHANGES_REQUESTED** [review]
There also is a console.log in query worker that needs to be removed
Honestly, the site was not faster for me at all could not even load all the counties in colorado
Also I assumed the grapher could graph any data from any constraint. maybe the origin of this pull is form an old branch, but also the grapher could not handle it when I tried to graph lots of future heatwave data

**Comment by Marc-9** [comment]
Also I need a quick run down on the graph querier code

**Comment by drg101** [comment]
@Marc-9 Update the branch, I resolved all of your issues. 
Try the county query now, I switched the file from the ~200mb to the ~30mb one. This PR won't fix all the issues with the query system, just all the client-facing ones.

**Comment by drg101** [comment]
I also don't really understand much of the graph code, but I think some optimizations will definitely be needed with this new system.

**Comment by drg101** [comment]
> Honestly, the site was not faster for me at all could not even load all the counties in colorado

@Marc-9 Make sure you are not using the heatwaves data as a benchmark, as it doesn't include much of Colorado. Use the risk index instead.

**Review by Marc-9 — APPROVED** [review]
All my issues and comments were addressed

File: src/iframe/js/library/autoQuery.js [hunk 6]
```diff
@@ -7,6 +7,7 @@
  */
 
 class AutoQuery {
+    static queryWorker = new SharedWorker('js/library/queryWorker.js', {name: "Auto query worker"}); //init querier
```
Comment (pierce-smith1) [inline src/iframe/js/library/autoQuery.js:10]: Out of curiosity, why is this static now? Were things breaking if multiple AutoQuery objects were made?

**Review by pierce-smith1 — APPROVED** [review]
This is insane

**Review by honeybadgerofdoom — APPROVED** [review]
Ludicrous speed indeed!

File: src/iframe/js/library/autoQuery.js (hunk 6 above)
Comment (drg101) [reply src/iframe/js/library/autoQuery.js:10]: There was just no reason to make more than one, as I am setting sessionID's with each request anyways.

**Pull request merged**